Bottom: 2d0059c2ad94cde8f11f53884aef1d9589c25234
Top:    1663ed1cfbbad3b8633a4f83692af86711344df8
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 21:50:00 +0000

hid/gtk: (GL) Disable double buffering of the GTK drawing widgets

 We already double buffer in GL, so we don't need GTK to "help" us here.
This might speed rendering up a little?


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 74929f9..875c388 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -807,6 +807,8 @@ ghid_init_drawing_widget (GtkWidget *widget, GHidPort *port)
 {
   render_priv *priv = port->render_priv;
 
+  GTK_WIDGET_UNSET_FLAGS (GTK_WIDGET (widget), GTK_DOUBLE_BUFFERED);
+
   gtk_widget_set_gl_capability (widget,
                                 priv->glconfig,
                                 NULL,
